sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -456,364 +465,82 @@ def size_of_files_in_glob(self, pattern, limit=None):
     return file_sizes
 
 
-# TODO: Consider using cStringIO instead of buffers and data_lists when reading.
-class GcsBufferedReader(object):
-  """A class for reading Google Cloud Storage files."""
-
-  def __init__(self,
-               client,
-               path,
-               mode='r',
-               buffer_size=DEFAULT_READ_BUFFER_SIZE):
-    self.client = client
-    self.path = path
-    self.bucket, self.name = parse_gcs_path(path)
-    self.mode = mode
-    self.buffer_size = buffer_size
+class GcsDownloader(Downloader):
+  def __init__(self, client, path, buffer_size):
+    self._client = client
+    self._path = path
+    self._bucket, self._name = parse_gcs_path(path)
+    self._buffer_size = buffer_size
 
     # Get object state.
-    self.get_request = (storage.StorageObjectsGetRequest(
-        bucket=self.bucket, object=self.name))
+    self._get_request = (storage.StorageObjectsGetRequest(
+        bucket=self._bucket, object=self._name))
     try:
-      metadata = self._get_object_metadata(self.get_request)
+      metadata = self._get_object_metadata(self._get_request)
     except HttpError as http_error:
       if http_error.status_code == 404:
-        raise IOError(errno.ENOENT, 'Not found: %s' % self.path)
+        raise IOError(errno.ENOENT, 'Not found: %s' % self._path)
       else:
-        logging.error('HTTP error while requesting file %s: %s', self.path,
+        logging.error('HTTP error while requesting file %s: %s', self._path,
                       http_error)
         raise
-    self.size = metadata.size
+    self._size = metadata.size
 
     # Ensure read is from file of the correct generation.
-    self.get_request.generation = metadata.generation
+    self._get_request.generation = metadata.generation
 
     # Initialize read buffer state.
     self.download_stream = cStringIO.StringIO()
```

Comment:
<!--new_thread; commit:79815e6df1d49c271a6c44e37fb2a3c46be0aed5; resolved:0-->
Can we add an underscore / make this private too?